src/form/time-input/TimeInput.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import classNames from "classnames";
+
+import Input from "../input/Input";
+import {DATE_FORMAT} from "../../core/utils/time/timeConstants";
+import {formatDateWithOptions, parseTime} from "../../core/utils/time/timeUtils";
+
+export interface TimeInputProps {
```

Comment:
Looks great. Thanks! 💯 

I have a few questions/requests about `TimeInputProps`
- I think we should get native input HTML attributes also. 
-  `testid` should be optional like we did on InputProps.
-  Should we replace `icon` prop to `leftIcon` and `rightIcon` props? I think it can be more flexible this way.
- In short, I think we can generate `TimeInputProps` from `InputProps` like this:
```tsx
export type TimeInputProps = Omit<
  InputProps,
  "localizationOptions" | "onChange" | "value"
> & {
  initialDateTime?: Date | null;
  onChange: (timeString: string) => void;
  value: string;
};
```

Here is the `InputProps` for reference:
```tsx
export type InputProps = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  "disabled" | "name" | "className"
> & {
  name: string;
  type?: InputTypes;
  testid?: string;
  leftIcon?: React.ReactNode;
  rightIcon?: React.ReactNode;
  isDisabled?: boolean;
  hasError?: boolean;
  customClassName?: string;
  onChange: React.ReactEventHandler<HTMLInputElement>;
  localizationOptions?: InputLocalizationOptions;
};
```